WordPress/gutenberg
# Remove "Advanced" toggle from LinkControl

## What?
Try removing the "Advanced" toggle and associated preference to simplify a bit. 

## Testing Instructions

1. Open a post or page.
2. Insert a heading block.
3. Add a link. 
4. Select "Edit" to further edit the link.
5. See changes. 

## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
|![CleanShot 2024-04-03 at 16 29 24](https://github.com/WordPress/gutenberg/assets/1813435/f77cbb0d-4ed3-452b-9d01-4144dc7def22)|![CleanShot 2024-04-03 at 16 24 10](https://github.com/WordPress/gutenberg/assets/1813435/afba5f4b-db32-4e0e-9f26-01af0e677786)|
